core/node/node_framework/src/service/mod.rs

```diff
@@ -145,30 +148,43 @@ impl ZkStackService {
             .collect();
 
         // Run the tasks until one of them exits.
-        // TODO (QIT-24): wrap every task into a timeout to prevent hanging.
-        let (resolved, idx, remaining) = self
+        let (resolved, resolved_idx, remaining) = self
             .runtime
             .block_on(futures::future::select_all(join_handles));
-        let task_name = tasks[idx].name.clone();
+        let resolved_task_name = tasks[resolved_idx].name.clone();
         let failure = match resolved {
             Ok(Ok(())) => {
-                tracing::info!("Task {task_name} completed");
+                tracing::info!("Task {resolved_task_name} completed");
                 false
             }
             Ok(Err(err)) => {
-                tracing::error!("Task {task_name} exited with an error: {err}");
+                tracing::error!("Task {resolved_task_name} exited with an error: {err}");
                 true
             }
             Err(_) => {
-                tracing::error!("Task {task_name} panicked");
+                tracing::error!("Task {resolved_task_name} panicked");
                 true
             }
         };
 
+        let remaining_tasks_with_timeout: Vec<_> = remaining
+            .into_iter()
+            .map(|task|
+                rt_handle.spawn(async move {
+                    if tokio::time::timeout(TASK_SHUTDOWN_TIMEOUT, task).await.is_err()
+                    {
+                        tracing::warn!(
```

Comment:
This would put a log line per each task that timed out, which looks weird given that we don't include the task name.

Reply:
as turned out, it is not trivial to add task name into logs here..

>This would put a log line per each task that timed out, which looks weird given that we don't include the task name.

at least we will know how much tasks reached timeout. What would u propose?

Reply:
See the comment below -- I don't propose including task names, I propose writing a log just once (and there I've described how you can do it).

Reply:
yeah...it will be at least cheaper